lib/statsd/instrument/batched_udp_sink.rb

```diff
@@ -64,9 +64,18 @@ def <<(datagram)
             # If the dispatcher thread is dead, we assume it is because
             # the process was forked. So to avoid ending datagrams twice
             # we clear the buffer.
-            @buffer.clear
-            @dispatcher_thread = Thread.new { dispatch }
+            # However if the main the main thread is dead, we won't be able
+            # to spawn a new thread, so we fallback to sending our datagram directly.
+            if Thread.main.alive?
```

Comment:
I don't believe `alive?` return false when a thread is in the `aborting` state unfortunately.  We probably want this to explicitly check `if Thread.main.status != "aborting"`

Reply:
🤦 Yeah, I totally skimmed the `Thread.main` part of that.  I just test it out and did confirm that if we avoid the raise when the main thread isn't alive every appears to work well:
```
  ensure
    puts "In ensure, Raising a new error"
    raise StandardError "In Rescue" if Thread.main.alive?
  end
  ```